Mathlib/Algebra/Group/UniqueProds.lean

```diff
@@ -196,36 +216,153 @@ attribute [to_additive UniqueSums] UniqueProds
-instance (priority := 100) Covariants.to_uniqueProds {A} [Mul A] [LinearOrder A]
-    [CovariantClass A A (· * ·) (· ≤ ·)] [CovariantClass A A (Function.swap (· * ·)) (· < ·)]
-    [ContravariantClass A A (· * ·) (· ≤ ·)] : UniqueProds A where
-      uniqueMul_of_nonempty {A} {B} hA hB :=
-        ⟨_, A.min'_mem ‹_›, _, B.min'_mem ‹_›, fun a b ha hb ab ↦
-        eq_and_eq_of_le_of_le_of_mul_le (Finset.min'_le _ _ ‹_›) (Finset.min'_le _ _ ‹_›) ab.le⟩
-#align covariants.to_unique_prods Covariants.to_uniqueProds
-#align covariants.to_unique_sums Covariants.to_uniqueSums
+/-- This instance asserts that if `A` has a right-cancellative multiplication, a linear order,
+  and multiplication is strictly monotone w.r.t. the second argument, then `A` has `UniqueProds`. -/
+@[to_additive _root_.Covariant.to_uniqueSums_right
+  "This instance asserts that if `A` has a right-cancellative addition, a linear order,
+  and addition is strictly monotone w.r.t. the second argument, then `A` has `UniqueSums`." ]
+instance (priority := 100) _root_.Covariant.to_uniqueProds_right {A} [Mul A] [IsRightCancelMul A]
+    [LinearOrder A] [CovariantClass A A (· * ·) (· < ·)] :
+    UniqueProds A where
+  uniqueMul_of_nonempty {A B} hA hB := by
+    obtain ⟨a0, b0, ha0, hb0, he⟩ := Finset.mem_mul.mp (Finset.max'_mem _ <| hA.mul hB)
+    refine ⟨a0, ha0, b0, hb0, fun a b ha hb he' => ?_⟩
+    obtain hl | rfl | hl := lt_trichotomy b b0
+    · refine ((he'.trans he ▸ mul_lt_mul_left' hl a).not_le <| Finset.le_max' _ (a * b0) ?_).elim
+      exact Finset.mem_mul.mpr ⟨a, b0, ha, hb0, rfl⟩
+    · exact ⟨mul_right_cancel he', rfl⟩
+    · refine ((he ▸ mul_lt_mul_left' hl a0).not_le <| Finset.le_max' _ (a0 * b) ?_).elim
+      exact Finset.mem_mul.mpr ⟨a0, b, ha0, hb, rfl⟩
+
+open MulOpposite in
+-- see Note [lower instance priority]
+/-- This instance asserts that if `A` has a left-cancellative multiplication, a linear order,
+  and multiplication is strictly monotone w.r.t. the first argument, then `A` has `UniqueProds`. -/
+@[to_additive _root_.Covariant.to_uniqueSums_left
+  "This instance asserts that if `A` has a left-cancellative addition, a linear order,
+  and addition is strictly monotone w.r.t. the first argument, then `A` has `UniqueSums`." ]
+instance (priority := 100) _root_.Covariant.to_uniqueProds_left {A} [Mul A] [IsLeftCancelMul A]
+    [LinearOrder A] [CovariantClass A A (Function.swap (· * ·)) (· < ·)] :
+    UniqueProds A :=
+let _ := LinearOrder.lift' (unop : Aᵐᵒᵖ → A) unop_injective
+let _ : CovariantClass Aᵐᵒᵖ Aᵐᵒᵖ (· * ·) (· < ·) :=
+{ elim := fun _ _ _ bc =>
+          have : StrictMono (unop (α := A)) := fun _ _ => id
+          mul_lt_mul_right' (α := A) bc (unop _) }
+of_mulOpposite _ Covariant.to_uniqueProds_right
+
+variable {G H : Type*} [Mul G] [Mul H]
+
+@[to_additive (attr := nontriviality, simp)]
+theorem of_subsingleton [Subsingleton G] : UniqueProds G :=
+  ⟨fun ⟨a, ha⟩ ⟨b, hb⟩ => ⟨a, ha, b, hb, UniqueMul.of_subsingleton⟩⟩
```

Comment:
What was the auto-generated name before #6751?

Reply:
`to_additive` worked on `UniqueProds`, I think because that was the namespace, so it actually gave it the correct name.  Btw, the `to_additive` PR is not merged into this one, so you can still check the naming in this one, if you want.